src/Driver/Driver.jl

```diff
@@ -143,53 +205,108 @@ function parse_commandline(custom_settings)
+                Settings.restart_from_num
+            end
+        end
         "--fix-rng-seed"
         help = "set RNG seed to a fixed value for reproducibility"
         action = :store_true
         "--log-level"
         help = "set the log level to one of debug/info/warn/error"
         metavar = "<level>"
         arg_type = String
-        default = "info"
+        default = if haskey(defaults, :log_level)
+            uppercase(defaults[:log_level])
+        else
+            uppercase(get(
+                ENV,
+                "CLIMATEMACHINE_SETTINGS_LOG_LEVEL",
+                Settings.log_level,
+            ))
+        end
         "--output-dir"
         help = "directory for output data"
         metavar = "<path>"
         arg_type = String
-        default = get(ENV, "CLIMATEMACHINE_OUTPUT_DIR", "output")
+        default = if haskey(defaults, :output_dir)
+            defaults[:output_dir]
+        elseif haskey(ENV, "CLIMATEMACHINE_OUTPUT_DIR")
+            get(ENV, "CLIMATEACHINE_OUTPUT_DIR")
+        else
+            get(ENV, "CLIMATEMACHINE_SETTINGS_OUTPUT_DIR", Settings.output_dir)
+        end
         "--integration-testing"
         help = "enable integration testing"
         action = :store_true
     end
-
+    # add custom cli argparse settings if provided
     if !isnothing(custom_settings)
         import_settings!(s, custom_settings)
     end
-
     return parse_args(s)
 end
 
+
 """
-    ClimateMachine.array_type()
+    ClimateMachine.cli(;arg_settings=nothing, init_driver=true, kwargs...)
 
-Return the array type used by ClimateMachine. This defaults to (CPU-based) `Array`
-and is only correctly set (based on choice from the command line, from
-an environment variable, or from experiment code) after `ClimateMachine.init()`
-is called.
+Initialize the ClimateMachine runtime with cli argument parsing.
+
+- Additional `ArgParseSetting` behavior can be injected into the default
+ClimateMachine `ArgParseSetting` cofiguration by setting the `arg_settings`
+keyword value.
+
+- Setting `init_driver=false` will set the ClimateMachine.Setting singleton
+values without initializing the ClimateMachine driver runtime.
+
+- ClimateMachine.init key value pairs can be supplied to overload
+default system defaults at runtime, default values will be merged with the
+parsed argument settings, with parsed cli argument values taking precedent
+over runtime defined default values.
+
+
+Returns a `Dict` containing parsed process ARGS values.
 """
-array_type() = Settings.array_type
+function cli(;
+    custom_settings::Union{Nothing, ArgParseSettings} = nothing,
+    init_driver::Bool = true,
+    kwargs...,
+)
+    kw_defaults = Dict{Symbol, Any}(kwargs)
+    parsed_args = parse_commandline(kw_defaults, custom_settings)
+    # we need to munge the parsed arg dict a bit as parsed arg keys
+    # and climatemachine initialization keywords are not 1:1
+    parsed_args["checkpoint_keep_one"] = !parsed_args["checkpoint_keep_all"]
```

Comment:
No idea why I did this. Let's change it so there isn't such an exception? In this file and in `Utilities/Callbacks/Callbacks.jl`.